File: Testes/ex1_slide_herancaMultipla/AudioLivro.cpp
#include "AudioLivro.hpp"

AudioLivro::AudioLivro(string titulo, string autor, string narrador, int numPags, int duracao)
    :Material(titulo, autor), Livro(titulo, autor, numPags), DVD(titulo, autor, duracao), narrador(narrador){};

void AudioLivro::descricao()
{
    Material::descricao();
    cout << "Número de páginas: " << numero_de_paginas
    <<"\nDuração: " << duracao << "min"
    << "\nNarrador: " << this->narrador <<  endl;
}

File: Laboratórios/Java/ex1_slide_herancaMutlipla/src/ex1_slide_herancaMutlipla/AudioLivro.java
package ex1_slide_herancaMutlipla;

public class AudioLivro extends Livro{
	protected DVD dvd;
	protected String narrador;
	
	public AudioLivro(String titutlo, String autor, int numPags, int duracao, String narrador) {
		super(titutlo, autor, numPags);
		this.dvd = new DVD(titutlo, autor, duracao);
		this.narrador = narrador;
	}

	@Override
	public void descricao() {
		super.descricao();
		System.out.printf("Duração: %d min\nNarrador: %s", dvd.duracao, narrador );
	}
	
	

}
